tecken/base/symbolstorage.py

```diff
@@ -82,16 +82,22 @@ class SymbolStorage:
 
     """
 
-    def __init__(self, urls, try_url_index=None):
-        self.urls = urls
-        self.backends = [
-            StorageBucket(url, try_symbols=(i == try_url_index))
-            for i, url in enumerate(urls)
+    def __init__(
+        self, upload_url: str, download_urls: list[str], try_url: Optional[str] = None
+    ):
+        self.upload_backend = StorageBucket(upload_url)
+        self.download_backends = [
+            StorageBucket(url) for url in download_urls if url != upload_url
         ]
-        self.try_url_index = try_url_index
+        if try_url is None:
+            self.try_backend = None
+        else:
+            self.try_backend = StorageBucket(try_url, try_symbols=True)
+        self.backends = [self.upload_backend, *self.download_backends]
```

Comment:
I included eveything in `backends` that used to be in `normal_storage.backends` before unifying the two `SymbolStorage` instances. The code in `_get()` decides whether to append `try_backend` based on the `try_storage` parameter to that function.

There were three reasons I did not include `try_backend` immediately:

1. In `_get()` we sometimes need to include it, and sometimes not. It seemed easier to append it if needed than to remove it when not needed.
2. The code in `libdockerflow.py` iterates over all backends and tests whether the corresponding buckets exist. The old code did not include the try storage in that test. I wanted to keep it that way, since the try storage lives in the same bucket anyway, so we don't need to test that separately.
3. There are some tests that initialize `SymbolStorage` without a `try_url`, in which case `try_storage` will be `None`. I could probably update these tests, but what I did was less effort, and it seems good enough.